ohttp-server/src/main.rs

```diff
@@ -171,7 +171,12 @@ fn fetch_maa_token(maa: &str) -> Res<String> {
 
 /// Retrieves the HPKE private key from Azure KMS.
 ///
-async fn get_hpke_private_key_from_kms(kms: &str, kid: u8, token: &str) -> Res<String> {
+async fn get_hpke_private_key_from_kms(
+    kms: &str,
+    kid: u8,
+    token: &str,
+    x_ms_request_id: &Uuid,
```

Comment:
Nit: can you change the variable name to request_id? The header name can of course be x-ms.